docs/pallas/quickstart.md

```diff
@@ -0,0 +1,230 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.7
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Pallas Quickstart
+
+Pallas is an extension to JAX that enables writing custom kernels for GPU and TPU. Pallas allows you to use the same JAX functions and APIs but operates at a *lower* level of abstraction.
+
+Specifically, Pallas requires users to think about memory access and how to divide up computations across multiple compute units in a hardware accelerator. On GPUs, Pallas lowers to Triton and on TPUs, Pallas lowers to Mosaic.
+
+Let's dive into some examples.
+
++++
+
+## Hello world in Pallas
+
+```{code-cell} ipython3
+import jax
+from jax.experimental import pallas as pl
+import jax.numpy as jnp
+import numpy as np
+```
+
+We'll first write the "hello world" in Pallas, a kernel that adds two vectors.
+
+```{code-cell} ipython3
+def add_vectors_kernel(x_ref, y_ref, o_ref):
+  x, y = x_ref[...], y_ref[...]
+  o_ref[...] = x + y
+```
+
+**`Ref` types**
+
+Let's dissect this function a bit. Unlike most JAX functions you've probably written, it does not take in `jax.Array`s as inputs and doesn't have any outputs. Instead it takes in *`Ref`* objects as inputs. Note that we also don't have any outputs but we are given an `o_ref`, which corresponds to the desired output.
+
+**Reading from `Ref`s**
+
+In the body, we are first reading from `x_ref` and `y_ref`, indicated by the `[...]` (the ellipsis means we are reading the whole `Ref`; alternatively we also could have used `x_ref[:]`). Reading from a `Ref` like this returns a `jax.Array`.
+
+**Writing to `Ref`s**
+
+We then write `x + y` to `o_ref`. Mutation has not historically been supported in JAX -- `jax.Array`s are immutable! `Ref`s are new (experimental) types that allow mutation under certain circumstances. We can interpret writing to a `Ref` as mutating its underlying buffer.
+
++++
+
+So we've written what we call a "kernel". It's a program that will be executed on an accelerator. How do we invoke it from a JAX computation? We use the `pallas_call` higher-order function.
+
+```{code-cell} ipython3
+@jax.jit
+def add_vectors(x: jax.Array, y: jax.Array) -> jax.Array:
+  return pl.pallas_call(add_vectors_kernel,
+                        out_shape=jax.ShapeDtypeStruct(x.shape, x.dtype)
+                        )(x, y)
+add_vectors(jnp.arange(8), jnp.arange(8))
+```
+
+`pallas_call` takes in a kernel function as input and some other metadata about it. Here we specify `out_shape`, an object that has a `.shape` and `.dtype` (or a list thereof).
```

Comment:
"some other metadata about it" doesn't add anything IMO. How about "`pallas_call` lifts the Pallas kernel function into an operation that can be called as part of a larger JAX program. But, to do so, it needs a few more details."

Reply:
Thanks that sounds much better.